lametro/views.py

```diff
@@ -734,6 +736,115 @@ class LAMetroArchiveSearch(TemplateView):
+            obj['agenda_link'] = obj['agenda_link'].split('\n')
+            obj['minutes_link'] = obj['minutes_link'].split('\n')
+
+        return filtered_historical_events
+
+    def _get_stored_events(self, start_datetime=None, end_datetime=None):
+        minutes = EventDocument.objects.filter(note__icontains='minutes').prefetch_related(Prefetch('links', to_attr='prefetched_links'))
+        agenda = EventDocument.objects.filter(note__icontains='agenda').prefetch_related(Prefetch('links', to_attr='prefetched_links'))
+        all_events = LAMetroEvent.objects.prefetch_related(Prefetch('documents', queryset=minutes, to_attr='minutes_document'))\
+                                         .prefetch_related(Prefetch('documents', queryset=agenda, to_attr='agenda_document'))
+        if start_datetime:
+            stored_events = all_events.filter(start_time__gt=start_datetime)\
+                                      .order_by('start_time')
+        else:
+            stored_events = all_events.order_by('start_time')
+        if end_datetime:
+            stored_events = stored_events.filter(start_time__lt=end_datetime)\
+                                         .order_by('start_time')
+        else:
+            stored_events = stored_events.filter(start_time__lt=timezone.now())\
+                                         .order_by('start_time')
+
+        structured_db_events = []
+        for event in stored_events:
+            stored_events_dict = {
+                'start_time': event.start_time.date(),
+                'meeting': event.name,
+                'minutes_link': [],
+                'agenda_link': [],
+            }
+            try:
+                minutes_link = event.minutes_document[0].prefetched_links[0].url
+                stored_events_dict['minutes_link'].append(minutes_link)
+            except IndexError:
+                pass
+
+            try:
+                agenda_link = event.agenda_document[0].prefetched_links[0].url
+                stored_events_dict['agenda_link'].append(agenda_link)
+            except IndexError:
+                pass
+
+            structured_db_events.append(stored_events_dict)
+
+        return structured_db_events
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        start_date_str = None
+        end_date_str = None
+        start_datetime = None
+        end_datetime = None
+        if 'minutes-from' in self.request.GET:
+            start_date_str = self.request.GET['minutes-from'] or None
+            if start_date_str:
+                start_datetime = datetime.datetime.strptime(start_date_str, '%m/%d/%Y')
+        if 'minutes-to' in self.request.GET:
+            end_date_str = self.request.GET['minutes-to'] or None
+            if end_date_str:
+                end_datetime = datetime.datetime.strptime(end_date_str, '%m/%d/%Y')
+
+        context['start_date'] = start_date_str
+        context['end_date'] = end_date_str
```

Comment:
If these are None, they show None in the date pickers.

<img width="722" alt="Screen Shot 2022-02-23 at 3 11 27 PM" src="https://user-images.githubusercontent.com/12176173/155409799-2c2670b3-8661-46ca-afdb-6647794fe5c1.png">

 How about an empty string if they aren't set?

Reply:
Ah, the None also breaks the date search! 

```
Environment:


Request Method: GET
Request URL: http://localhost:8001/minutes/?minutes-from=None&minutes-to=02%2F01%2F2022

Django Version: 2.2.25
Python Version: 3.6.14
Installed Applications:
('django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.gis',
 'haystack',
 'opencivicdata.core',
 'opencivicdata.legislative',
 'lametro',
 'councilmatic_core',
 'adv_cache_tag',
 'debug_toolbar',
 'captcha')
Installed Middleware:
('debug_toolbar.middleware.DebugToolbarMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.security.SecurityMiddleware')



Traceback:

File "/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py" in inner
  34.             response = get_response(request)

File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py" in _get_response
  115.                 response = self.process_exception_by_middleware(e, request)

File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py" in _get_response
  113.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/usr/local/lib/python3.6/site-packages/django/views/generic/base.py" in view
  71.             return self.dispatch(request, *args, **kwargs)

File "/usr/local/lib/python3.6/site-packages/django/views/generic/base.py" in dispatch
  97.         return handler(request, *args, **kwargs)

File "/usr/local/lib/python3.6/site-packages/django/views/generic/list.py" in get
  157.         context = self.get_context_data()

File "/app/lametro/views.py" in get_context_data
  821.                 start_datetime = datetime.datetime.strptime(start_date_str, '%m/%d/%Y')

File "/usr/local/lib/python3.6/_strptime.py" in _strptime_datetime
  565.     tt, fraction = _strptime(data_string, format)

File "/usr/local/lib/python3.6/_strptime.py" in _strptime
  362.                          (data_string, format))

Exception Type: ValueError at /minutes/
Exception Value: time data 'None' does not match format '%m/%d/%Y'
```

I think an empty string should resolve this. 👍